### PR TITLE
Add fieldNames option to catalyst output

### PR DESCRIPTION
**Pull request by ghost**
Add fieldNames and onlyPlotSpecifiedFieldNames options to catalyst output.
If onlyPlotSpecifiedFieldNames flag is equal to 1, then we only plot the fields listed in `fieldNames`. Otherwise, we plot all the fields with the required `plotLevel`, plus the fields listed in `fieldNames`.

### CODE REVIEW

**Comment by ghost** [comment]
@CharlesGueunet please review

File: src/coreComponents/fileIO/Outputs/CatalystOutput.cpp [hunk 1]
```diff
@@ -368,6 +368,9 @@ struct CatalystOutput::CatalystInternals {
   std::string adiosConfig = "";
   std::string channelName = "fullfield";
   std::string sstFileName = "gs.bp";
+  int onlyPlotSpecifiedFieldNames = 0;
```
Comment (vollia) [inline src/coreComponents/fileIO/Outputs/CatalystOutput.cpp]:
```suggestion
  integer onlyPlotSpecifiedFieldNames = 0;
```

File: src/coreComponents/fileIO/Outputs/CatalystOutput.cpp [hunk 2]
```diff
@@ -530,6 +556,71 @@ void CatalystOutput::cleanup(real64 const time_n, integer const cycleNumber,
   this->internal->initialized = false;
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////
+bool CatalystOutput::isFieldDefined(const std::string name)
```
Comment (vollia) [inline src/coreComponents/fileIO/Outputs/CatalystOutput.cpp]: can be reference to avoid copy

**Comment by vollia** [comment]
LGTM after small remark

**Pull request merged**